plugins/CoreAdminHome/Controller.php

```diff
@@ -255,11 +255,22 @@ public function trackingCodeGenerator()
     }
 
     /**
-     * Shows the "Track Visits" checkbox.
+     * Shows the "Track Visits" checkbox - iFrame (deprecated)
      */
     public function optOut()
     {
-        return $this->optOutManager->getOptOutView()->render();
+        return $this->optOutManager->getOptOutViewIframe()->render();
+    }
+
+    /**
+     * Shows the Javascript opt out
+     *
+     * @return string
+     * @throws Exception
+     */
+    public function optOutJS(): string
+    {
+        return $this->optOutManager->getOptOutJS();
```

Comment:
This method currently seems to start the session. This shouldn't be required as the method only serves JS code.
Guess we can avoid the session start by adding an additional check here:
https://github.com/matomo-org/matomo/blob/df0fb6290ce496b9520e2062293aeb32c68978af/core/FrontController.php#L705-L709

Reply:
Good spot :+1: I've added an exclusion to the front controller.

Reply:
Works 👍  Another thing that just came to my mind was, if we should send some additional headers here. Like the correct content type or some caching related headers. 🤔

Reply:
That makes sense, caching would prevent any server side changes being applied via the inline settings data :+1: I've added a content type header for 'application/javascript' and set caching to 'no-store'.